src/realm/list.cpp

```diff
@@ -219,53 +219,42 @@ void Lst<ObjKey>::do_remove(size_t ndx)
 }
 
 template <>
-void Lst<ObjKey>::clear()
+void Lst<ObjKey>::do_clear()
 {
-    update_if_needed();
-    auto sz = Lst<ObjKey>::size();
-
-    if (sz > 0) {
-        auto origin_table = m_obj.get_table();
-        TableRef target_table = m_obj.get_target_table(m_col_key);
-
-        if (Replication* repl = m_obj.get_replication())
```

Comment:
Can you explain why this isn't needed anymore?

Reply:
It is done in the outer function